app/src/main/java/com/example/nidoqueue/controller/RequestManager.java

```diff
@@ -1,64 +1,95 @@
 package com.example.nidoqueue.controller;
 
+import android.annotation.SuppressLint;
+import android.app.Application;
 import android.content.Intent;
 import android.provider.Settings;
 
 import com.example.nidoqueue.DataCalc;
-import com.example.nidoqueue.model.Database;
-import com.example.nidoqueue.model.Experiment;
 import com.example.nidoqueue.R;
-import com.example.nidoqueue.model.User;
 import com.example.nidoqueue.activity.AbstractActivity;
 import com.example.nidoqueue.activity.MainActivity;
 import com.example.nidoqueue.activity.SearchActivity;
 import com.example.nidoqueue.activity.SignInActivity;
 import com.example.nidoqueue.activity.UserProfileActivity;
+import com.example.nidoqueue.model.Experiment;
+import com.example.nidoqueue.model.User;
+import com.google.firebase.firestore.FirebaseFirestore;
 
-import static com.example.nidoqueue.controller.UserControl.contextManager;
-
-public class RequestManager {
+public class RequestManager extends Application {
 
     // Singleton pattern
-    private static final RequestManager requestManager = new RequestManager();
-    private RequestManager(){}
```

Comment:
I think we should leave these lines unchanged, I think this is the correct way to make a singleton class.  We create the instance using static so that it exists at a class level.  No other classes need to call this constructor, so that is why it is declared private.  The constructor is called once during runtime.  This post on tutorials point also uses this format in their example for an android singleton. https://www.tutorialspoint.com/how-to-use-singleton-class-in-android